packages/core/src/types.ts

```diff
@@ -0,0 +1,58 @@
+declare global {
+  interface GlobalContext {
+    __merkur__: Merkur;
+  }
+}
+
+export type AnyFn = (...args: unknown[]) => unknown;
+export type AnyObj = Record<string, unknown>;
+// export type Widget = AnyObj; // TODO
+
+export type RegisterFunction = ({
+  name,
+  version,
+  createWidget,
+}: {
+  name: string;
+  version: string;
+  createWidget: CreateFunction;
+}) => void;
+
+export type Plugin = {
+  setup: (merkur: Merkur) => void;
+};
+
+export type WidgetProperties = {
+  name: string;
+  version: string;
+};
+
+export type Merkur = {
+  $in: {
+    widgets: Array<Merkur>;
+    widgetFactory: Record<string, CreateFunction>;
+  };
+  $external: AnyObj;
+  $dependencies: AnyObj;
+  $plugins: Array<() => Plugin>;
+  register?: RegisterFunction;
```

Comment:
register is always defined https://github.com/mjancarik/merkur/blob/master/packages/core/src/merkur.js#L54